new resource "azurerm_maintenance_assignment" (#6713)
terraform-providers/terraform-provider-azurerm: website/docs/r/maintenance_assignment_virtual_machine.html.markdown

  location            = azurerm_resource_group.example.location
  size                = "Standard_F2"
  admin_username      = "adminuser"
  network_interface_ids = [
    azurerm_network_interface.example.id,
  ]

  admin_ssh_key {
    username   = "adminuser"
    public_key = file("~/.ssh/id_rsa.pub")
  }

  os_disk {
    caching              = "ReadWrite"
    storage_account_type = "Standard_LRS"
  }

  source_image_reference {
    publisher = "Canonical"
    offer     = "UbuntuServer"
    sku       = "16.04-LTS"
    version   = "latest"
  }
}

resource "azurerm_maintenance_configuration" "example" {
  name                = "example-mc"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
  scope               = "All"
}

resource "azurerm_maintenance_assignment_virtual_machine" "example" {
  location                     = azurerm_resource_group.example.location
  maintenance_configuration_id = azurerm_maintenance_configuration.example.id
  virtual_machine_id           = azurerm_linux_virtual_machine.example.id
}
```

## Argument Reference

The following arguments are supported:

> * `location` - (Required) Specified the supported Azure location where the resource exists. Changing this forces a new resource to be created.

Reviewer: typo
```suggestion
* `location` - (Required) Specifies the supported Azure location where the resource exists. Changing this forces a new resource to be created.
```

> * `maintenance_configuration_id` - (Required) Specified the id of Maintenance Configuration Resource. Changing this forces a new resource to be created.

Reviewer: typo
```suggestion
* `maintenance_configuration_id` - (Required) Specifies the ID of the Maintenance Configuration Resource. Changing this forces a new resource to be created.
```

> * `virtual_machine_id` - (Required) Specified the Virtual Machine ID where the Maintenance Configuration will be assigned to. Changing this forces a new resource to be created.

Reviewer: typo
```suggestion
* `virtual_machine_id` - (Required) Specifies the Virtual Machine ID which the Maintenance Configuration will be assigned to. Changing this forces a new resource to be created.
```
